test/test_generation_mode.py

```diff
@@ -0,0 +1,83 @@
+from dataclasses import dataclass
+from pathlib import Path
+
+import pytest
+
+from simple_parsing.wrappers.field_wrapper import ArgumentGenerationMode, NestedMode
+from . import TestSetup
+
+
+@dataclass
+class ModelOptions:
+  path: str
+  device: str
+
+
+@dataclass
+class ServerOptions(TestSetup):
+  host: str
+  port: int
+  model: ModelOptions
+
+
+def assert_as_expected(options: ServerOptions):
```

Comment:
Would you mind defining an `expected` variable in the global scope instead, and doing `assert <thing> == expected` instead of calls to this?
```python
expected = ServerOptions(host="myserver", port=80, model=ModelOptions(path="a_path", device="cpu")
```

Reply:
It makes sense. Done here: https://github.com/lebrice/SimpleParsing/pull/117/commits/367b5bc3fbb3c512fda27d2492a0fc491c43dc76